### ISSUE TITLE
Feature orientations are different on different platforms

### ISSUE THREAD

**Issue by tgxs002** [colmap/colmap#951]
Thank you for your great work, it helped me a lot! 

I was trying to register new images to an existing model built from COLMAP. However, the new features extracted can not match with the old features. Later I found that the previous model was built on Ubuntu while I was doing experiments on Windows, and this turned out to be the problem.

The bug I found is that the 'orientation' column of table 'keypoints' in the database is different between Ubuntu and Windows. For the same keypoint, (here by 'same' I mean they are close to each other in the picture), the orientation of feature extracted on Ubuntu and Windows add up to be close to 2 * PI, which is around 6.28. This happens across most versions ranging from 1.0 to 3.2, as far as I have tested.

I know in the latest version it is using affinity instead of scale and orientation, and this may no longer be a problem, so I'm just reporting it here in case it is useful. If so, I'm happy to provide more details.

**Comment by ahojnnes** [comment]
Thanks, the reason might be that you used CPU vs GPU based feature extraction on the different platforms. Seems like this is caused by inconsistent conventions between SiftGPU and VLFeat - something that could be fixed easily probably.

**Comment by tgxs002** [comment]
I agree this might be caused by different conventions. But the strange thing is that on Windows, the orientation is always consistent, no matter using CPU or GPU. So, it's more likely to be caused by difference of OS rather than code.

**Comment by ahojnnes** [comment]
That's odd. No idea why this would happen and I can only guess different behavior of geometric functions in the standard library of some ifdef's causing different behavior in VLFeat/SiftGPU.
